crates/gitbutler-branch-actions/src/branch.rs

```diff
@@ -425,58 +425,65 @@ pub fn get_branch_listing_details(
         .filter_map(Result::ok)
         .collect();
     let repo = ctx.repository();
-    let branches = list_branches(ctx, None, Some(branch_names.clone()))?;
-    let default_target = ctx
-        .project()
-        .virtual_branches()
-        .get_default_target()
-        .context("failed to get default target")?;
-    let mut enriched_branches = Vec::new();
-
-    let default_local_branch =
-        repo.find_branch(default_target.branch.branch(), git2::BranchType::Local)?;
-    let default_branch = default_local_branch.upstream()?;
-    let head_commit = default_branch.get().peel_to_commit()?;
+    let branches = list_branches(ctx, None, Some(branch_names))?;
+
+    let (default_target_upstream_commit_id, default_target_commit_id) = {
+        let default_target = ctx
+            .project()
+            .virtual_branches()
+            .get_default_target()
+            .context("failed to get default target")?;
+        let default_local_branch =
+            repo.find_branch(default_target.branch.branch(), git2::BranchType::Local)?;
+        let default_branch = default_local_branch.upstream()?;
```

Comment:
This would probably return the local tracking branch, `refs/remotes/origin/main`, right?

Reply:
i think this should give you `refs/heads/main`

Reply:
That is a bit unexpected as I thought `default_target.branch.branch()` is `main` and `upstream()` should then be something else - the only upstream branch I can think of is the local tracking branch. But once there is a test, I think this should become clear.
We are probably looking at the difference between diffing against what was last fetched (with the local tracking branch `refs/remotes/origin/main` updated), and the local state that we would rebase against, `refs/heads/main`.

Reply:
`upstream()` does return the local tracking branch, e.g. typically `refs/remotes/origin/main`. And interestingly, the `sha` field seems to be the merge-base between the local tracking branch and HEAD at the time the target is computed. 
This strikes me like an error source as if that drifts, but maybe that's not a problem as long as it recomputes it when switching to the integration branch.

Reply:
> i think this should give you refs/heads/main
Okay i am actually not sure of my own answer. I will take a look tomorrow to see i can give some more helpful feedback.

Reply:
By now I have made sense of it all, and the actual `upstream()` (i.e. local tracking branch) is used for virtual branches that aren't in the workspace, or other branches. Otherwise, the last-seen (and stored) location of the local tracking branch is used, as applied virtual branches are always assumed to be applied on top of that local tracking branch state.

Running `update` on it would then re-apply them on top of the latest location of the local tracking branch.